db/migrate/20151011151130_create_course_user_achievements.rb

```diff
@@ -0,0 +1,13 @@
+class CreateCourseUserAchievements < ActiveRecord::Migration
+  def change
+    create_table :course_user_achievements do |t|
+      t.references :course_user
+      t.references :achievement, foreign_key: { references: :course_achievements }
+      t.datetime :obtained_at, null: false
```

Comment:
@lowjoel This field has different meaning with `created_at`, `created_at` is the time when user really get the achievement, but the `obtained_at` should be set to the time when student submit his assessment.  Thus TA's performance won't affect student's leaderboard ranking. 

Maybe I should give you an example:

S submits his mission on Monday
T grades S's submission on Tuesday and S get the achievement on Tuesday
S's achievement's `obtained_at` should be set to Monday